Create contributing to ros (#6918)
ros2/ros2_documentation: source/Contact.rst

Contact
=======

> Contact us the ROS community to ask for support, offer help, or suggest ways we can improve.

Reviewer: ```suggestion
To contact the ROS community to ask for support, offer help, or suggest ways we can improve.
```

> Contact us the ROS community to ask for support, offer help, or suggest ways we can improve.

Reviewer 2: Remove the 'us'.

```suggestion
Contact the ROS community to ask for support, offer help, or suggest ways we can improve.
```

This article introduces the ways in which you can contact us.

**Area: community | content-type: about | experience: beginner, intermediate, expert**

.. contents:: Table of Contents
   :local:

Summary
-------

You can contact the ROS community in the following ways:

> * `Stack Exchange <https://robotics.stackexchange.com/>`__: Ask for support or discuss an issue

Reviewer 2: "Issue" implies a GitHub issue to me. Suggest "problem" instead.

Reviewer 3: ```suggestion
* `Stack Exchange <https://robotics.stackexchange.com/>`__: Ask for support or discuss a problem
```

* Issue tracker: Submit an issue or suggest an enhancement: <Placeholder link to new "Submitting an issue" article>

* :ref:`Discourse discussion boards <discussion-boards>`: Share ideas or discuss best practices

* `Zulip chat <https://openrobotics.zulipchat.com/Recent>`_
   (QUESTION FOR THE REVIEWER: What is the purpose for this one?

>    Is it meant more for people who are involved in developing ROS rather than for users with questions?)

Reviewer: Yes, both, it is basically a Discord server. If you want to chat with someone in real time it is probably your best bet. Questions should be directed to the `General` or the `ROS General` topics.

>    Is it meant more for people who are involved in developing ROS rather than for users with questions?)

Reviewer 3: ```suggestion
* `Zulip chat <https://openrobotics.zulipchat.com/Recent>`_
   Chat with other community members, including the maintainers, about ROS, and see project work happening in real-time
```

> * `Email <mailto:ros@osrfoundation.org>`_: Contact us directly if you have a private, sensitive, or security issue

Reviewer 2: I believe this should be ros@openrobotics.org. @gbiggs ?

Reviewer 2: I thought we had a different email for security reports, but I only see a `security@osrfoundation.org` address and I don't think that's the one we want. @gbiggs ?

> * `Email <mailto:ros@osrfoundation.org>`_: Contact us directly if you have a private, sensitive, or security issue

Reviewer 3: ```suggestion
* `Email <mailto:ros@openrobotics.org>`_: Contact us directly if you have a private or sensitive issue

* `Email (security matters) <mailto:security@openrobotics.org`_: Contact us directly if you have found a security problem in ROS
```

.. _asking-for-support:

Asking for support
------------------

Need help troubleshooting your system?
First, search `Robotics Stack Exchange <https://robotics.stackexchange.com/>`__ to see if others have had similar issues, and if their solution works for you.

If not, ask a new question on `Robotics Stack Exchange <https://robotics.stackexchange.com/>`__.

> Make sure to add tags, at the very least the ``ros2`` tag and the distro version you are running, e.g. ``{DISTRO}``.

Reviewer 3: ```suggestion
Make sure to add tags, at the very least the ``ros`` tag and the distro version you are running, e.g. ``{DISTRO}``.
```

We are moving to just saying "ROS" everywhere, now that ROS 1 is a year past EOL.

> If your question is related to the documentation, add a tag like ``docs``, or more specifically, ``tutorials``.

Reviewer 2: To me this reads like the suggestion is to use a `tutorials` tag on all docs questions. Perhaps: "If your question is related to the documentation, add a ``docs`` tag and, optionally, a tag for the specific area of the documentation like ``tutorials``."

> If your question is related to the documentation, add a tag like ``docs``, or more specifically, ``tutorials``.

Reviewer 3: ```suggestion
If your question is related to the documentation, add a tag like ``docs``, and, optionally, a tag for the specific area of the documentation, such as ``tutorials`` or ``concepts``.
```

Please don't contact the developers/maintainers directly.

> The community can't see question or answer(s) not asked or answered publicly.

Reviewer 2: Suggested rephrase: "The community can't benefit from private answers and questions. Keep your questions out in the public to help others solve their problems."

> The community can't see question or answer(s) not asked or answered publicly.

Reviewer 3: ```suggestion
The community can't benefit from private questions and answers. Keep your questions in public to help others who encounter similar problems.
```

Open Source development works best when the entire community participates in discussions and helps to answer questions.

> It's better to send all questions to `Robotics Stack Exchange <https://robotics.stackexchange.com/>`__ and report all issues to the issue tracker.

Reviewer 3: ```suggestion
```

.. note:: If you can provide support to others, please do: <Placeholder link to new Contributing to ROS article>

.. _reporting-an-issue:

Reporting an issue or suggesting an enhancement
-----------------------------------------------

> If you identify issues, have suggestions for enhancements, or have a question specific to one package, you can raise an issue on GitHub.

Reviewer 2: Suggestion: replace `raise` with `report`. Not only is the latter more commonly used, it matches the wording of the soon-to-be-linked article.

> If you identify issues, have suggestions for enhancements, or have a question specific to one package, you can raise an issue on GitHub.

Reviewer 3: ```suggestion
If you identify an issue in or have a suggestion for an enhancement to a package, you can report an issue on GitHub.
```

<Placeholder link to new "Reporting an issue" article>

> .. note:: If you have a private, sensitive, or security issue, email us directly at `ros@rosfoundation.org <mailto:ros@srosfoundation.org>`__.

Reviewer: Security issues use a different e-mail address, namely [security@openrobotics.org](mailto:security@openrobotics.org). 

Also I don't believe the current address is correct.

> .. note:: If you have a private, sensitive, or security issue, email us directly at `ros@rosfoundation.org <mailto:ros@srosfoundation.org>`__.

Reviewer 3: ```suggestion
.. note:: If you have a private or sensitive issue, email us directly at `ros@openrobotics.org <mailto:ros@openrobotics.org>`__. For reporting security issues, email us at `security@openrobotics.org <mailto:security@openrobotics.org>`__
```

.. _discussion-boards:

Discussing ideas and best practices
-----------------------------------

To start a discussion with other ROS community members, visit the official `Open Robotics Discourse <https://discourse.openrobotics.org/>`_.

> Content on the Discourse should be high-level.

Reviewer 3: ```suggestion
Content on Discourse should be discussions and announcements.
```

> It's not a place to get *questions* about code answered, but it would be suitable to start a conversation about best practices or improving standards.

Reviewer 3: ```suggestion
It is not a place to get *questions* about code answered, but it would be suitable to start a conversation about new ideas, best practices, and improving standards.
```

> Discussions about ROS development and plans are also happening on the `Open Robotics Discourse' ROS category <https://discourse.openrobotics.org/c/ros/111>`_.

Reviewer 3: ```suggestion
Discussions about ROS are in the `ROS category <https://discourse.openrobotics.org/c/ros/111>`_.
```

Participating in these discussions is an important way to have a say on how different features of ROS will work and be implemented.

The diverse community behind the ROS ecosystem is one of its greatest assets.
We encourage all members of the ROS community to participate in these design discussions so that we can leverage the experience of community members, and keep

> the varied use cases of ROS in mind.

Reviewer 3: ```suggestion
We encourage all members of the ROS community to participate in these design discussions so that we can leverage the experience of community members, and keep the varied use cases of ROS in mind when design decisions are made.
```

Offering help
-------------

> If you are in a position to support others, share your knowledge and help fellow developers work through challenges.

Reviewer 3: ```suggestion
If you are in a position to support others, share your knowledge and help fellow users work through challenges.
```

<Placeholder link to new "Contributing to ROS" article>

Contacting us privately
-----------------------

If you'd like to contact us privately, for example, if your question contains information sensitive to your organization or project, or if it's regarding a

> security issue, you can email us directly at `ros@rosfoundation.org <mailto:ros@rosfoundation.org>`__.

Reviewer: This e-mail address is not correct. I think ros@openrobotics.org is what you want.

Reviewer 2: Also, security issues should go to the `security@` email address as Kat mentioned above.

> security issue, you can email us directly at `ros@rosfoundation.org <mailto:ros@rosfoundation.org>`__.

Reviewer 3: ```suggestion
If you'd like to contact us privately, for example, if your question contains information sensitive to your organization or project, you can email us directly at `ros@openrobotics.org <mailto:ros@openrobotics.org>`__
If your message is regarding a security issue, email `security@openrobotics.org <mailto:security@openrobotics.org>`__.
```

   This also applies to cross-posting.
   Try to pick the forum which you think matches best and ask there.
   If you are referred to a new forum, provide a link to the old discussion.

* On `Robotics Stack Exchange <https://robotics.stackexchange.com/>`__ , you can edit your question to provide more details.
   The more details that you include in your question, the easier it is for others to help you find your solution which makes it more likely for you to get a response.

* It's considered bad form to list your personal deadlines.
   Community members answering questions also have them.

* Do not beg for help.
   If there is someone willing and able to help with your problem, you usually get a response.
   Asking for faster answers will mostly have a negative effect.

* Do not add unrelated content to posts.
   The content of posts should be focused on the topic at hand and not include unrelated content.
   Content, links, and images unrelated to the topic are considered spam.

* For commercial posts, see also `this discussion <https://discourse.openrobotics.org/t/sponsorship-notation-in-posts-on-ros-org/2078>`_.

* Minimize references to content behind pay walls.

>    The content posted on Open Robotics Discourse and Robotics Stack Exchange should "generally" be free and open to all users.

Reviewer 2: Generally doesn't need scare quotes here.

>    The content posted on Open Robotics Discourse and Robotics Stack Exchange should "generally" be free and open to all users.

Reviewer 3: ```suggestion
   The content posted on Open Robotics Discourse and Robotics Stack Exchange should be free and open to all users.
```

* `Stack Exchange <https://robotics.stackexchange.com/>`__: Ask for support or discuss a problem

* Issue tracker: Submit an issue or suggest an enhancement: <Placeholder link to new "Submitting an issue" article>

* :ref:`Discourse discussion boards <discussion-boards>`: Share ideas or discuss best practices

* `Zulip chat <https://openrobotics.zulipchat.com/Recent>`_: Chat to other ROS community members in real time in the ``General`` or ``ROS General`` topics.

* `Email <mailto:ros@openrobotics.org>`_: Contact us directly if you have a private or sensitive issue

* `Email <mailto:security@openrobotics.org>`_: Contact us directly if you have a security concern

.. _asking-for-support:

Asking for support
------------------

Need help troubleshooting your system?
First, search `Robotics Stack Exchange <https://robotics.stackexchange.com/>`__ to see if others have had similar problems, and if their solution works for you.

If not, ask a new question on `Robotics Stack Exchange <https://robotics.stackexchange.com/>`__.
Make sure to add tags, at the very least the ``ros`` tag and the distro version you are running, e.g. ``{DISTRO}``.

> If your question relates to the documentation, add a ``docs`` tag.make

Reviewer 2: If your question relates to the documentation, add a ``docs`` tag.
